### DISCUSSION TITLE
can a group of dependencies be excluded by tox for a specific task?

### DISCUSSION THREAD

**Discussion by allynt** [tox-dev/tox#2542] [Q&A] [answered]
I am using `tox` & `poetry` & `pytest` & `sphinx`.

My "pyproject.toml" looks like a bit like this:

```
[tool.poetry]
name = "my-app"
readme = "README.md"
packages = [{include = "my_app"}]

[tool.poetry.dependencies]
python = "^3.8"
django = "^3.2 || ^4.0"

[tool.poetry.group.dev.dependencies]
tox = "*"
pytest = "*"

[tool.poetry.group.docs.dependencies]
sphinx = "*"

[build-system]
requires = ["poetry-core>=1.0.0"]
build-backend = "poetry.core.masonry.api"
```

And my "tox.ini" looks a bit like this:

```
[tox]
envlist = py{38,310}-django{32,40,41}
isolated_build = true

[testenv]
allowlist_externals = 
    poetry
    make
deps =
    pytest
commands_pre = 
    poetry install --no-root --sync
commands =
    poetry run pytest ./tests

[testenv:docs]
deps = 
    sphinx
changedir = docs
commands = 
    make clean
    make html SPHINXOPTS="-v"
```

Running `poetry run tox -e docs` works - it builds the documentation.  And running `poetry run tox` works - it runs my tests.  _But_, it also installs all the dependencies listed under "[tool.poetry.group.docs.dependencies]" and those are not needed for running any tests.  

Is it possible to exclude those?

**Comment by jugmac00** [comment, marked as answer]
Afaik not with tox 3, but with the upcoming version 4 you can use labels, and then group environments.

https://tox.wiki/en/rewrite/config.html#labels

Still, it won't have a mechanism for exclusion, but rather one for inclusion.

So you could then run `tox -m all-but-docs`.

Please note, there is no ETA for tox 4 yet.

**Reply by allynt** [reply]
Okay.  Thanks.
